bindings/profiler.cc

```diff
@@ -85,73 +82,142 @@ NAN_METHOD(StopSamplingHeapProfiler) {
+  std::vector<CpuProfileNode::LineTick> entries(hitLineCount);
+  if (hasLines && node->GetLineTicks(&entries[0], hitLineCount)) {
+    children = Nan::New<Array>(count + entries.size());
+    js_node->Set(Nan::New<String>("hitCount").ToLocalChecked(),
+                 Nan::New<Integer>(0));
+    for (const CpuProfileNode::LineTick entry : entries) {
+      Local<Object> js_node_hit = Nan::New<Object>();
+      js_node_hit->Set(Nan::New<String>("name").ToLocalChecked(),
+                       Nan::New<String>("").ToLocalChecked());
+      js_node_hit->Set(Nan::New<String>("scriptName").ToLocalChecked(),
+                       node->GetScriptResourceName());
+      js_node_hit->Set(Nan::New<String>("scriptId").ToLocalChecked(),
+                       Nan::New<Integer>(node->GetScriptId()));
+      js_node_hit->Set(Nan::New<String>("lineNumber").ToLocalChecked(),
+                       Nan::New<Integer>(entry.line));
+      js_node_hit->Set(Nan::New<String>("columnNumber").ToLocalChecked(),
+                       Nan::New<Integer>(0));
+      js_node_hit->Set(Nan::New<String>("hitCount").ToLocalChecked(),
+                       Nan::New<Integer>(entry.hit_count));
+      js_node_hit->Set(Nan::New<String>("children").ToLocalChecked(),
+                       Nan::New<Array>(0));
+      children->Set(index++, js_node_hit);
+    }
+  } else {
+    children = Nan::New<Array>(count);
+  }
+
+  // Add nodes corresponding to functions called by the node's function.
   for (int32_t i = 0; i < count; i++) {
-    children->Set(i, TranslateTimeProfileNode(node->GetChild(i)));
+    children->Set(index++,
+                  TranslateTimeProfileNode(node->GetChild(i), hasLines));
   }
-  js_node->Set(Nan::New<String>("children").ToLocalChecked(),
-    children);
+
+  js_node->Set(Nan::New<String>("children").ToLocalChecked(), children);
   return js_node;
 }
 
-Local<Value> TranslateTimeProfile(const CpuProfile* profile) {
+Local<Value> TranslateTimeProfile(const CpuProfile* profile, bool hasLines) {
   Local<Object> js_profile = Nan::New<Object>();
   js_profile->Set(Nan::New<String>("title").ToLocalChecked(),
-    profile->GetTitle());
-  js_profile->Set(Nan::New<String>("topDownRoot").ToLocalChecked(),
-    TranslateTimeProfileNode(profile->GetTopDownRoot()));
+                  profile->GetTitle());
+  js_profile->Set(
+      Nan::New<String>("topDownRoot").ToLocalChecked(),
+      TranslateTimeProfileNode(profile->GetTopDownRoot(), hasLines));
   js_profile->Set(Nan::New<String>("startTime").ToLocalChecked(),
-    Nan::New<Number>(profile->GetStartTime()));
+                  Nan::New<Number>(profile->GetStartTime()));
   js_profile->Set(Nan::New<String>("endTime").ToLocalChecked(),
-    Nan::New<Number>(profile->GetEndTime()));
+                  Nan::New<Number>(profile->GetEndTime()));
   return js_profile;
 }
 
 NAN_METHOD(StartProfiling) {
-  Local<String> name = info[0].As<String>();
+  if (info.Length() != 2) {
```

Comment:
I find it useful to have a comment as the first line indicating the JS function signature so that reading the code is easier. Based on the code below, I expect the comment to be
```c++
// startProfiling(name: String, includeLineInfo: boolean)
```

Reply:
Done.